app/helpers/application_helper.rb

```diff
@@ -22,4 +22,11 @@ def render_flash_message(flash_value)
     end
   end
 
+  def user_has_membership_application?
+    current_user.try(:membership_applications).try(:any?) && current_user.try(:membership_applications).try(:last).try(:persisted?)
+  end
```

Comment:
Agreed -- this logic shouldn't be in the views.  @thesuss  had it in the navigation menus (I just pulled it out into a helper).  The menus need to be refactored so that all of this kind of logic is addressed better.